indra/assemblers/pybel/assembler.py

```diff
@@ -626,16 +626,16 @@ def _get_evidence(evidence):
         )
 
     annotations = {
-        'source_hash': evidence.get_source_hash(),
+        'source_hash': {evidence.get_source_hash(): True},
     }
     if evidence.source_api:
-        annotations['source_api'] = evidence.source_api
+        annotations['source_api'] = {evidence.source_api: True}
     if evidence.source_id:
-        annotations['source_id'] = evidence.source_id
+        annotations['source_id'] = {evidence.source_id: True}
     for key, value in evidence.epistemics.items():
         if key == 'direct':
             continue
-        annotations[key] = value
+        annotations[key] = {value: True}
```

Comment:
The value here is None or a list. As list is unhashable, we can replace it with a tuple. I am not sure how we'd want to handle None. {None: True}?

Reply:
so the type of `evidence.epistemics` is `Mapping[str, Union[None, List[str]]]`?

Reply:
I believe the type is `Mapping[str, Union[None, bool, List[str]]]`, so it can be a boolean as well. @bgyori can you confirm this?

Reply:
Yes, that's right, values in the epistemics dict are sometimes lists (e.g., from Eidos), which would error here, we just don't have a test to cover that yet. Just to make this more concrete, an example entry is
```
{'hedgings': ['might', 'consider']}
```
Should we skip these annotations (given that they aren't critical for PyBEL analysis) or should we make a sorted tuple out of the value?

Reply:
Actually, I think `annotations[key] = {v: True for v in value}` as @cthoyt suggested could work. I'm not sure we need special handling for None and bool (i.e., `if key == 'direct' or value is None`), those are valid dict keys, or am I missing something?

Reply:
Adding `None`'s will be an issue, so `if value is None: continue` is important. The `key == 'direct'` check is skipped because that part of the epistemics dictionary is used elsewhere to determine the edge type, so it would be redundant here.

Reply:
I made the update in https://github.com/sorgerlab/indra/pull/1187/files#diff-01513c4320c875fa208bd896b195470920e0bf2415486994cacbb3077eff7ace, but I think it would be good to include a test that covers this

Reply:
Full solution (that's robust to lists/scalars) and corresponding tests added in 615774d63 and 47183edd1

Reply:
Thanks @cthoyt, will merge once Travis is done!